src/mapeo-project.js

```diff
@@ -749,6 +761,410 @@ export class MapeoProject extends TypedEmitter {
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {string} [options.lang]
+   * @returns {Promise<string>} The full path that the file was exported at
+   */
+  async exportGeoJSONFile(
+    exportFolder,
+    { observations = true, tracks = true, lang } = {}
+  ) {
+    const fileName = await this[kGeoJSONFileName](observations, tracks)
+    const filePath = path.join(exportFolder, fileName)
+    const source = this[kExportGeoJSONStream]({ observations, tracks, lang })
+    const sink = createWriteStream(filePath)
+    await pipelinePromise(source, sink)
+
+    return filePath
+  }
+
+  /**
+   * @param {Attachment} attachment
+   * @returns {Promise<null | BlobId>}
+   */
+  async #tryGetBlobId(attachment) {
+    // Audio must not have variants
+    for (const variant of VARIANT_EXPORT_ORDER) {
+      const blobId = buildBlobId(attachment, variant)
+      const entry = await this.#blobStore.entry(blobId)
+      if (!entry) continue
+      return blobId
+    }
+
+    return null
+  }
+
+  /**
+   * @param {ZipArchive} archive
+   * @param {Object} [options={}]
+   * @param {boolean} [options.observations=true] Whether observations should be exported
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {boolean} [options.attachments=true] Whether all attachments for observations should be exported
+   * @param {string} [options.lang]
+   * @returns {Promise<void>}
+   */
+  async #exportToArchive(
+    archive,
+    { observations = true, tracks = true, attachments = true, lang } = {}
+  ) {
+    // GeoJSON
+    const geoJSONFileName = await this[kGeoJSONFileName](observations, tracks)
+    const seenAttachments = new Map()
+    const geoJSONStream = this[kExportGeoJSONStream]({
+      observations,
+      tracks,
+      lang,
+      seenAttachments,
+    })
+
+    // @ts-expect-error
+    await archive.entry(geoJSONStream, { name: geoJSONFileName })
+
+    const missingAttachments = []
+    // Attachments
+    if (attachments) {
+      const mediaFolder = this.#exportPrefix('Media') + '/'
+      for (const attachment of seenAttachments.values()) {
+        const blobId = await this.#tryGetBlobId(attachment)
+        if (blobId === null) {
+          missingAttachments.push(attachment)
+          continue
+        }
+
+        const stream = this.#blobStore.createReadStream(blobId)
+        const name = mediaFolder + blobId.variant + '/' + attachment.name
+
+        // @ts-expect-error
+        await archive.entry(stream, { name })
+      }
+
+      if (missingAttachments.length) {
+        this.#l.log(`Found missing attachments during export`)
+        const missingContent = missingAttachments
+          .map((attachment) => JSON.stringify(attachment))
+          .join('\n')
+
+        await archive.entry(missingContent, {
+          name: mediaFolder + 'missing.ndjson',
```

Comment:
We need to validate this with some user research, but I think in many cases a human-readable file is going to be most useful, because the user story for export is someone without programming skills exporting data to use in GIS tools or another system, and they will want to review what data is missing from the export to check whether they need to coordinate users to sync data to get all the missing data. We maybe need further work to identify what information is needed in this list of missing data that will help a user identify what needs to be done to get the data.